python/sglang/srt/managers/schedule_batch.py

```diff
@@ -247,8 +247,11 @@ def __init__(
         self.embedding = None
 
         # Constrained decoding
-        self.regex_fsm: RegexGuide = None
+        self.regex_fsm: Optional[RegexGuide] = None
         self.regex_fsm_state: int = 0
+        self.regex_bnf: Optional[GrammarMatcher] = None
```

Comment:
Build a better abstraction for this.
Merge `regex_fsm`, `regex_fsm_state`, `regex_bnf` into one variable and the scheduler should not know it is outlines or xgrammar.